match/lib/match.rb

```diff
@@ -29,4 +29,21 @@ def self.environments
   def self.enterprise?
     ENV["MATCH_FORCE_ENTERPRISE"]
   end
+
+  def self.type_is_enterprise?(type)
+    Match.enterprise? && type != "development"
```

Comment:
Previously there were two different checks: `type != "development"` and `type == "enterprise"`. I chose to go with the more permissive one.